torchbenchmark/util/distributed/core_model/trainer.py

```diff
@@ -93,7 +93,10 @@ def measure(self):
                 grad_params = {}
                 for name, param in self.benchmark.model.named_parameters():
                     if param.requires_grad:
-                        grad_params[name + ".grad"] = param.grad.cpu()
+                        if param.grad is not None:
+                            grad_params[name + ".grad"] = param.grad.cpu()
```

Comment:
is this happening in fsdp bc parameters are the views i hacked?

if so, does this mean params are gonna bypass accuracy check in fsdp bc of this?

Reply:
good point, let me take a closer look into why this is happening.

Reply:
I verified that this is happening in both eager and dynamo+inductor; i.e. all the params for which `param.grad == None`

cc @awgu do you have any idea why this would be? i.e. for some parameters, `param.requires_grad == True` but `param.grad is None`. This doesn't happen in DDP for these models.

Reply:
this is happening bc of `use_orig_params` being set to true.  i think.  

my fsdp enablement PR puts orig_params views into module .parameters field, so  at least aot_autograd can find them and wrap them in faketensors.  But as they are views, I am not sure if their requires_grad field is set to true. or if that is only true on the underlying flat buffer...

Reply:
For any original parameter `param` that is not in the rank's shard of the parameters, `param.grad = None`. (Otherwise, if `param` has at least some presence in the rank's shard, then `param.grad` is the 1D flattened shard of the gradient.) However, `param.requires_grad == True` because it preserved the original `requires_grad` and makes sense because the parameter logically should receive a gradient (ignoring the sharding).

Is this a problem/blocker?

Reply:
Actually, I wonder if we can set `param.requires_grad = False` because the unsharded parameter is actually a different Python variable (that is a plain `Tensor`). Either way, let me know if these parameters having `requires_grad == True` but having `.grad = None` is a problem/blocker.

Reply:
@awgu not a problem - but will this strategy (iterate through parameters and check the ones with `param.requires_grad and param.grad is not None`) find all the gradients? or are there other gradients that I need to check that won't be caught in this check?